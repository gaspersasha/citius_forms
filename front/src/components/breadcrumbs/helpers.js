export const breadcrumbTitles = {
  '/cars': 'Cars for sale',
  '/cars/new-cars': 'New Cars in Stock',
  '/cars/part-exchange': 'Get A Quote For Your Car',
  '/vans': 'Vans For Sale',
  '/leasing': 'Cars To Lease',
  '/motoring-services': 'Motoring Services',
  '/how-it-works': 'How It Works',
  '/how-it-works/finance-options': 'Finance Options',
  '/how-it-works/warranty-insurance': 'Warranty & Insurance',
  '/how-it-works/part-exchange': 'Part Exchange',
  '/how-it-works/cancellation-returns': 'Cancellations & Returns',
  '/how-it-works/faq': 'FAQ',
  '/how-it-works/about': 'About Us',
  '/how-it-works/contact-us': 'Contact us',
  '/account': 'My Account',
  '/account/my-cars': 'My Cars',
  '/account/details': 'My Details',
  '/account/orders': 'My Orders',
  // TODO: Temporary duplicated for hidden routes
  '/account/my-details': 'My Details',
  '/account/my-orders': 'My Orders',
  '/part-exchange': 'Your Current Cars',
  '/finance-thank-you': 'Thank You',
  // ...others
};
