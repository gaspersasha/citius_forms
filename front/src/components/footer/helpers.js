import { CONTACTS, URL } from '~constants';

const { FACEBOOK, TWITTER, LINKEDIN, INSTAGRAM, PRIVACY, AUTOVIA_ORIGIN } = URL;

export const links = [
  { title: 'Reservation', href: '/cars' },
  { title: 'Used Cars', href: '/cars' },
  { title: 'New Cars in Stock', href: '/cars/new-cars' },
  { title: 'Vans', href: '/vans' },
  { title: 'How It Works', href: '/how-it-works' },
  { title: 'FAQ', href: '/how-it-works/faq' },
  { title: 'About Us', href: '/how-it-works/about' },
  { title: 'As seen in the press', href: '/press' },
  { title: "Buyer's Guides", href: '/buyers-guides' },
  { title: 'News', href: '/news' },
  { title: 'Advice', href: '/advice' },
  { title: 'Complaints', href: '/how-it-works/contact-us#complaints' },
  {
    title: 'Initial Disclosure Document',
    href: '/cars/1224/initial-disclosure-document',
  },
  {
    title: 'Work For Us',
    href: `${AUTOVIA_ORIGIN}/careers/search/?department=&brand=BuyaCar&location=&typeOfEmployment=`,
    target: '_blank',
  },
  { title: 'Dealer Sales', href: '/dealer-sales' },
  {
    title: 'Privacy Policy',
    href: PRIVACY,
    target: '_blank',
  },
  {
    title: 'Cookie Policy',
    href: `${AUTOVIA_ORIGIN}/cookies-policy`,
    target: '_blank',
  },
  { title: 'Sitemap', href: '/sitemap' },
];

export const footerSocials = [
  {
    title: 'facebook',
    href: FACEBOOK,
    target: '_blank',
    rel: 'noreferrer',
  },
  {
    title: 'twitter',
    href: TWITTER,
    target: '_blank',
    rel: 'noreferrer',
  },
  {
    title: 'linkedin',
    href: LINKEDIN,
    target: '_blank',
    rel: 'noreferrer',
  },
  {
    title: 'email',
    href: `mailto:${CONTACTS.EMAIL}`,
  },
  {
    title: 'instagram',
    href: INSTAGRAM,
    target: '_blank',
    rel: 'noreferrer',
  },
];
