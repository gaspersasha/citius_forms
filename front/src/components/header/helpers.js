export const pages = [
  {
    path: '/cars',
    name: 'Cars',
    withSubheader: true,
    subAlign: 'start',
    subpaths: [
      ['/cars', 'Used Cars'],
      ['/cars/new-cars', 'New Cars in Stock'],
      ['/part-ex', 'Sell my car'],
    ],
  },
  { path: '/goreserve', name: 'Reservation From' },
  { path: '/part-ex', name: 'Part Ex Form' },
  { path: '/motoring-services', name: 'Motoring Services' },
  {
    path: '/how-it-works',
    name: 'How It Works',
    withSubheader: false,
    subAlign: '',
    subpaths: [
      ['/how-it-works', 'Used Cars'],
      ['/how-it-works/finance-options', 'Finance Options'],
      ['/how-it-works/warranty-insurance', 'Warranty & Insurance'],
      ['/how-it-works/part-exchange', 'Part Exchange'],
      ['/how-it-works/cancellation-returns', 'Cancellations & Returns'],
      ['/how-it-works/faq', 'FAQ'],
      ['/how-it-works/about', 'About Us'],
      ['/how-it-works/contact-us', 'Contact us'],
    ],
  },
];

export const accountPage = {
  path: '/account',
  name: 'My Account',
  withSubheader: true,
  subAlign: 'end',
  subpaths: [
    ['/account/my-cars', 'My Cars'],
    ['/account/details', 'My Details'],
    ['/account/orders', 'My Orders'],
    ['/account/part-exchange', 'My Part-exchange'],
  ],
};
