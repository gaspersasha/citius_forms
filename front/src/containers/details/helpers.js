import { VALIDATION_TYPES } from '~constants';
import { getFormFiledProperties, composeFormState } from '~utils';

export const composeDetailsState = (state) => {
  const additional = {
    emailConfirm: getFormFiledProperties(state.email, VALIDATION_TYPES.EMAIL),
    password: getFormFiledProperties('', VALIDATION_TYPES.PASSWORD),
    passwordConfirm: getFormFiledProperties('', VALIDATION_TYPES.PASSWORD),
  };

  return composeFormState(state, additional);
};
