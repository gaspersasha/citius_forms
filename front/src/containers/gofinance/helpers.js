import { capitalize, GTM_EVENT, GTM as gtm } from '~utils';

const GTM_MAP = {
  start: 'ssFinanceApplicationStart',
  drivingLicenceType: 'ssDrivingLicense',
  maritalStatus: 'ssMaritalStatus',
  validUKPassport: 'ssPassport',
  dependants: 'ssDependents',
  residentialStatus: 'ssResidentialStatus',
  addressConfirm: 'ssAddressConfirmation',
  addressChange: 'ssAddressChange',
  residenceYears: 'ssResidedLength',
  employmentType: 'ssEmploymentType',
  employer: 'ssEmployer',
  postCode: 'ssEmployerAddress',
  employmentYears: 'ssEmploymentLength',
  annualSalary: 'ssAnnualIncome',
  accountNumber: 'ssBankDetails',
  yearsAtBank: 'ssBankLength',
};

// Self Service Journey
export const sendGTM = (some) => {
  GTM_EVENT(GTM_MAP[some]);
};

export const handleGTM = (state, keys) => {
  Object.keys(state).forEach((key) => {
    const pos = keys.indexOf(key);

    if (pos > -1) sendGTM(keys[pos]);
  });
};

export const GTM = (path) => gtm(path, 'finance', 'Finance your car');

export const formatOption = (type = '') => {
  const word = type.split('_').join(' ').toLowerCase();

  return capitalize(word);
};

export const getYearsAtJobLabel = (status, type) => {
  if (status !== 'unemployed')
    return 'How many years have you been working at this address?';

  const option = formatOption(type).toLowerCase();

  switch (type) {
    case 'UNEMPLOYED':
    case 'RETIRED':
    case 'UNABLE_TO_WORK':
      return `How long have you been ${option}?`;
    case 'STUDENT':
    case 'HOMEMAKER':
      return `How long have you been a ${option}?`;
    default:
      return `How many years have you spent as a ${option}?`;
  }
};

export const truncateData = ({
  addresses,
  details,
  employments,
  bankDetails,
  ...rest
}) => rest;
