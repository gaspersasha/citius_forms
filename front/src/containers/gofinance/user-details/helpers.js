export const getCreditScoreLevel = (score) => {
  if (!score || typeof score !== 'number') return null;
  if (score < 50) return 'Bad';
  if (score < 239) return 'Very poor';
  if (score < 320) return 'Poor';
  if (score < 410) return 'Fair';
  if (score < 470) return 'Good';

  return 'Excellent';
};
