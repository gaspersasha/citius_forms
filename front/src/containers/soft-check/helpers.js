import {
  CarCity,
  CarConvertible,
  CarCoupe,
  CarEstate,
  CarHatchback,
  CarMPV,
  CarSaloon,
  CarSUV,
} from '~assets/svg';

export const bodyStylesData = [
  {
    name: 'Hatchback',
    link: '/cars/family-cars/hatchbacks',
    icon: CarHatchback,
  },
  {
    name: 'Estate',
    link: '/cars/family-cars/estate-cars',
    icon: CarEstate,
  },
  {
    name: 'SUV',
    link: '/cars/family-cars/suvs',
    icon: CarSUV,
  },
  {
    name: 'Saloon',
    link: '/cars/family-cars/saloon-cars',
    icon: CarSaloon,
  },
  {
    name: 'Coupe',
    link: '/cars/sports-cars/coupes',
    icon: CarCoupe,
  },
  {
    name: 'Convertible',
    link: '/cars/sports-cars/convertibles',
    icon: CarConvertible,
  },
  {
    name: 'MPV',
    link: '/cars/family-cars/mpvs',
    icon: CarMPV,
  },
  {
    name: 'City car',
    link: '/cars?f[0]=body:city-car&f[1]=vehicle_type:car',
    icon: CarCity,
  },
];
