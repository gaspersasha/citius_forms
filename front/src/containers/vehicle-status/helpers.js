export const steps = {
  SEARCH: {
    id: 'search',
    text: 'Search',
  },
  FINANCE: {
    id: 'finance',
    text: 'Finance',
  },
  ORDER: {
    id: 'order',
    text: 'Order',
  },
  VERIFY: {
    id: 'verify',
    text: 'Verify & confirm',
  },
  VEHICLE_PREP: {
    id: 'verify-prep',
    text: 'Vehicle prep',
  },
  DELIVERY: {
    id: 'delivery',
    text: 'Delivery',
  },
};

export const subSteps = {
  // search (finance flow)
  START: 'start',
  // finance (finance flow)
  SOFT: 'soft',
  APPLIED: 'applied',
  PART_EX: 'part-ex',

  // order (finance flow)
  PENDING_NO_PAID: 'pending-no-paid',
  PENDING_PAID: 'pending-paid',
  APPROVED_NO_PAID: 'approved-no-paid',
  APPROVED_PAID: 'approved-paid',
  SECCI_SENT: 'secci-sent', // common sub-step for finance and cash
  // order (cash flow)
  SECURE_VEHICLE: 'secure-vehicle',
  // verify (finance)
  GETTING_REQUIREMENTS: 'getting-requirements',
  LINK_SENT: 'link-sent',
  DOCS_SENT: 'docs-sent',
  CONDITION_REPORT: 'condition-report', // common sub-step for finance and cash
  // delivery (finance flow and cash flow - common)
  TO_BE_BOOKED: 'to-be-booked',
  BOOKED: 'booked',
  CONFIRMED: 'confirmed',
};

export const financeFlowSteps = [
  steps.SEARCH,
  steps.FINANCE,
  steps.ORDER,
  steps.VERIFY,
  steps.DELIVERY,
];
export const cashFlowSteps = [steps.ORDER, steps.VEHICLE_PREP, steps.DELIVERY];

export const getProgress = (subStep, isFinanceFlow) => {
  if (isFinanceFlow) {
    // finance flow
    switch (subStep) {
      // search
      case subSteps.START:
        return 50;
      // finance
      case subSteps.SOFT:
        return 25;
      case subSteps.APPLIED:
        return 50;
      case subSteps.PART_EX:
        return 75;
      // order
      case subSteps.PENDING_NO_PAID:
        return 17;
      case subSteps.PENDING_PAID:
        return 33;
      case subSteps.APPROVED_NO_PAID:
        return 50;
      case subSteps.APPROVED_PAID:
        return 67;
      case subSteps.SECCI_SENT:
        return 83;
      // verify
      case subSteps.GETTING_REQUIREMENTS:
        return 20;
      case subSteps.LINK_SENT:
        return 40;
      case subSteps.DOCS_SENT:
        return 60;
      case subSteps.CONDITION_REPORT:
        return 80;
      // delivery
      case subSteps.TO_BE_BOOKED:
        return 33;
      case subSteps.BOOKED:
        return 66;
      case subSteps.CONFIRMED:
        return 100;
      default:
        return 100;
    }
  }

  // cash flow
  switch (subStep) {
    // order
    case subSteps.SECURE_VEHICLE:
      return 33;
    case subSteps.SECCI_SENT:
      return 66;
    // vehicle prep
    case subSteps.CONDITION_REPORT:
      return 50;
    // delivery
    case subSteps.TO_BE_BOOKED:
      return 33;
    case subSteps.BOOKED:
      return 66;
    case subSteps.CONFIRMED:
      return 100;
    default:
      return 100;
  }
};
