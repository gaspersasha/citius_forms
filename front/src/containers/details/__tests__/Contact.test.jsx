import React from 'react';
import { getFormFieldsNames } from '~utils';
import { Contact } from '..';
import { composeDetailsState } from '../helpers';

const fields = composeDetailsState(global.mockData.details);

describe('<Contact />', () => {
  it('renders without errors', () => {
    const wrapper = shallow(
      <Contact
        names={getFormFieldsNames(fields)}
        fields={fields}
        handleChange={jest.fn()}
      />
    );

    expect(wrapper).toMatchSnapshot();
  });
});
