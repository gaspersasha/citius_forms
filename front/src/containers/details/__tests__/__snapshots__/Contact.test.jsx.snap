// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Contact /> renders without errors 1`] = `
<Fragment>
  <Typography
    align="left"
    className="subheading subheadMargin"
    color="dark"
    type="p"
  >
    2. Date of birth and contact
  </Typography>
  <InputDate
    className="input"
    errorMessage=""
    id="dateOfBirth"
    label="Date of birth"
    name="dateOfBirth"
    onChange={[MockFunction]}
    placeholder="DD/MM/YY"
    required={true}
    status="default"
    subTitle="You must be at least 18 to apply for finance"
    useErrorBorder={true}
    value="11/11/2000"
  />
  <InputPhone
    className="input"
    errorMessage=""
    id="phone"
    label="Telephone number"
    name="phone"
    onChange={[MockFunction]}
    onKeyPress={null}
    placeholder="e.g 07712345678"
    required={true}
    status="default"
    subTitle="To keep you updated on your purchase"
    useErrorBorder={true}
    value="02057685678"
  />
  <InputPhone
    className="input"
    errorMessage=""
    id="alternativePhone"
    label="Alternative telephone number"
    name="alternativePhone"
    onChange={[MockFunction]}
    onKeyPress={null}
    placeholder="e.g. 07898765432"
    required={false}
    status="default"
    subTitle=""
    useErrorBorder={true}
    value="02057685678"
  />
</Fragment>
`;
